docs/templates-howto.md

```diff
@@ -1,130 +1,160 @@
+| packagePlaceholder | String | yes | it is possible to define a name which will be substituted with the app-name in the whole project directory during generation. | `"packagePlaceholder": "starwit",` |
+| imageUrl | String | no | the url of the template preview image | `"imageUrl": "https://raw.githubusercontent.com/starwit/reacthook-spring-template/main/screenshot.png",` |
+| templateFiles | Array | no | a list of template files for generating domain-specific code |   `"templateFiles": [...]` |
 
-#### 3. Configure Apptemplate
+## Add Templates
+### The Usage of Freemarker
 
 You can configure your templatefiles via [Freemarker](https://freemarker.apache.org/). It is used as templating language for configuring the path to code templates as well as in the template files itself. Use the properties in the [Template Configuration](#template-configuration) section to find out which properties you can use in the templates.
+### Add Template Files
 
-#### 4. Create a App
-
-The next step is the creation of the project itself. Create an new project in ProjectBuilder and choose your git-project and apptemplate and enter a name and description.
-
-#### 5. Add Domains and Attributes
-
-After successfully saving the general project settings, you can add your domain objects and attributes to the app.
-
-#### 6. Download your App
+In order to have domain-specific files, you additionally have to add them to `template-config.json` (as alsready displayed in the example above).
 
-Now, everything is prepared and you can download your new app.
+| property | type | madatory | description | example |
+|---|---|---|---|---|
+|fileName | String | yes | the name of the generated file. You can use [freemarker substitution mechanism](#freemarker-substitutions-for-templatepath) | `"fileName": "${entity.name}Entity.java",` |
+|templatePath | String | yes | the location ot the template based on the directory `generator-templates` | `"templatePath": "persistence/Entity.ftl",` |
+|targetPath | String | yes | the location of the target folder based on project directory | `"targetPath": "application/src/main/resources/"` |
+| append | Boolean | no | defines, if the generated content should be appended to the existing file (yes) or added as new file (no). Default value is no. | `"append": no,` |
+|category | String | no | You can use it for getting a better overview of yout template configurations | `"category": "ENTITY"` |
 
-# How to Configure Apptemplates
+### Freemarker Substitutions for TemplatePath and TargetPath
 
-### **project**
+The following substitutions are valid in template path:
 
-The project-object contains general information and can be used to configure the paths for templatefiles-sets saved as apptemplate.
+* `app` (or `project`<sup>depricated</sup>): all properties of the app defined in ProjectBuilder. See AppDto definition on https://pb.starwit.de/ljprojectbuilder/swagger-ui.html or definitions below
```

Comment:
I would generally remove the deprecated parts of this documentation. The user should build templates with the not deprecated parts...

```suggestion
* `app` (or `project`<sup>deprecated</sup>): all properties of the app defined in ProjectBuilder. See AppDto definition on https://pb.starwit.de/ljprojectbuilder/swagger-ui.html or definitions below
```